src/stores/notifications/ListNotificationState.ts

```diff
@@ -38,24 +42,25 @@ export class ListNotificationState extends NotificationState {
     public setRooms(rooms: Room[]) {
         // If we're only concerned about the tile count, don't bother setting up listeners.
         if (this.byTileCount) {
-            this.rooms = rooms;
+            this.roomIds = this.getRoomIdsFromRooms(rooms);
             this.calculateTotalState();
             return;
         }
 
-        const oldRooms = this.rooms;
-        const diff = arrayDiff(oldRooms, rooms);
-        this.rooms = rooms;
```

Comment:
I tend to shy away from shallow cloning whenever possible to avoid confusion later down the line. But I reckon this is not likely to be an issue in this case? (this is not supposed to sound aggressive at all, I'm just simply curious)

But yeah, if we can get away with shallow cloning, that makes this change a lot simpler, will change.

Reply:
Right, in this case a shallow clone is actually the behavior that we want (since as you said, otherwise the array could be changed outside the class). The previous approach of `.map`ing the rooms to their room IDs seemed to me that it was serving the same purpose as a shallow clone, just in a more roundabout way.